pkg/plugin/standalone/v1/logger.go

```diff
@@ -0,0 +1,161 @@
+	hclog.Debug:   zerolog.DebugLevel,
+	hclog.Info:    zerolog.InfoLevel,
+	hclog.Warn:    zerolog.WarnLevel,
+	hclog.Error:   zerolog.ErrorLevel,
+}
+
+type hcLogger struct {
+	name   string
+	logger zerolog.Logger
+}
+
+var _ hclog.Logger = (*hcLogger)(nil)
+
+func (h *hcLogger) Log(level hclog.Level, msg string, args ...interface{}) {
+	zlevel := hclogZerologLevelMapping[level]
+	h.applyArgs(h.logger.WithLevel(zlevel), args).Msg(msg)
+}
+
+func (h *hcLogger) Trace(msg string, args ...interface{}) {
+	h.applyArgs(h.logger.Trace(), args).Msg(msg)
+}
+
+func (h *hcLogger) Debug(msg string, args ...interface{}) {
+	h.applyArgs(h.logger.Debug(), args).Msg(msg)
+}
+
+func (h *hcLogger) Info(msg string, args ...interface{}) {
+	h.applyArgs(h.logger.Info(), args).Msg(msg)
+}
+
+func (h *hcLogger) Warn(msg string, args ...interface{}) {
+	h.applyArgs(h.logger.Warn(), args).Msg(msg)
+}
+
+func (h *hcLogger) Error(msg string, args ...interface{}) {
+	h.applyArgs(h.logger.Error(), args).Msg(msg)
+}
+
+func (h *hcLogger) IsTrace() bool {
+	return h.isLevel(zerolog.TraceLevel)
+}
+
+func (h *hcLogger) IsDebug() bool {
+	return h.isLevel(zerolog.DebugLevel)
+}
+
+func (h *hcLogger) IsInfo() bool {
+	return h.isLevel(zerolog.InfoLevel)
+}
+
+func (h *hcLogger) IsWarn() bool {
+	return h.isLevel(zerolog.WarnLevel)
+}
+
+func (h *hcLogger) IsError() bool {
+	return h.isLevel(zerolog.ErrorLevel)
+}
+
+func (h *hcLogger) ImpliedArgs() []interface{} {
+	panic("implement me") // TODO
+}
+
+func (h *hcLogger) With(args ...interface{}) hclog.Logger {
+	c := h.logger.With()
+	for i := 0; i < len(args)-1; i += 2 {
+		key := args[i].(string) // keys must be strings
+		val := args[i+1]
+		c.Interface(key, val)
+	}
+	h.logger = c.Logger()
+	return h
+}
+
+func (h *hcLogger) Name() string {
+	return h.name
+}
+
+func (h *hcLogger) Named(name string) hclog.Logger {
+	hCopy := *h
+	hCopy.name = strings.TrimLeft(h.name+" "+name, " ")
+	return &hCopy
+}
+
+func (h *hcLogger) ResetNamed(name string) hclog.Logger {
+	hCopy := *h
+	hCopy.name = name
+	return &hCopy
+}
+
+func (h *hcLogger) SetLevel(level hclog.Level) {
+	zlevel := hclogZerologLevelMapping[level]
+	h.logger = h.logger.Level(zlevel)
+}
+
+func (h *hcLogger) StandardLogger(opts *hclog.StandardLoggerOptions) *stdlog.Logger {
+	var prefix string
+	if h.name != "" {
+		prefix = fmt.Sprintf("%s: ", h.name)
+	}
+
+	return stdlog.New(h.StandardWriter(opts), prefix, 0)
+}
+
+func (h *hcLogger) StandardWriter(opts *hclog.StandardLoggerOptions) io.Writer {
+	return h.logger
+}
+
+func (h *hcLogger) applyArgs(e *zerolog.Event, args []interface{}) *zerolog.Event {
+	if h.name != "" {
+		// add plugin path name
+		e.Str(log.PluginNameField, h.name)
+	}
+	for i := 0; i < len(args)-1; i += 2 {
+		key := args[i].(string) // keys must be strings
+		switch key {
+		case "timestamp":
+			// skip timestamps, they are added by our internal logger
+			continue
+		case "@module":
+			key = log.ComponentField
+		}
+
+		val := args[i+1]
```

Comment:
Similar to above: if we have an odd number of params, this would panic?

Reply:
In this case it wouldn't panic, because of how the for loop condition is set up, but the last parameter would get lost in the matrix :neckbeard:

Reply:
Changed it so we log all params, even if it's an odd number.